controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -65,47 +97,460 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+		r.log.Error(
+			err,
+			"Failed to update CephClient.",
+			"CephClient",
+			klog.KRef(r.cephClientProvisioner.Namespace, r.cephClientProvisioner.Name),
+		)
+		return err
+	}
+
+	phase := ""
+	if r.cephClientProvisioner.Status != nil {
+		phase = string(r.cephClientProvisioner.Status.Phase)
+	}
+
+	r.setCephResourceStatus(r.cephClientProvisioner.Name, "CephClient", phase, nil)
+
+	return nil
+}
+
+func (r *StorageClassClaimReconciler) reconcileCephClientCephFSNode() error {
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephClientNode, func() error {
+		if err := r.own(r.cephClientNode); err != nil {
+			return err
+		}
+
+		addStorageRelatedAnnotations(r.cephClientNode, r.storageClassClaim.Name, "cephfs", "node")
+		r.cephClientNode.Spec = rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "allow r",
+				"mgr": "allow rw",
+				"osd": "allow rw tag cephfs *=*",
+				"mds": fmt.Sprintf("allow rw path=/volumes/%s", r.cephFilesystemSubVolumeGroup.Name),
+			},
+		}
+		return nil
+	})
+
+	if err != nil {
+		r.log.Error(
+			err,
+			"Failed to update CephClient.",
+			"CephClient",
+			klog.KRef(r.cephClientNode.Namespace, r.cephClientNode.Name),
+		)
+		return err
+	}
+
+	phase := ""
+	if r.cephClientNode.Status != nil {
+		phase = string(r.cephClientNode.Status.Phase)
+	}
+
+	r.setCephResourceStatus(r.cephClientNode.Name, "CephClient", phase, nil)
+
+	return nil
+}
+
+func (r *StorageClassClaimReconciler) setCephResourceStatus(name string, kind string, phase string, cephClients map[string]string) {
+
+	cephResourceSpec := r.cephResourcesByName[name]
+
+	if cephResourceSpec == nil {
+		cephResourceSpec = &ocsv1alpha1.CephResourcesSpec{
+			Name:        name,
+			Kind:        kind,
+			CephClients: cephClients,
+		}
+		r.storageClassClaim.Status.CephResources = append(r.storageClassClaim.Status.CephResources, cephResourceSpec)
+		r.cephResourcesByName[name] = cephResourceSpec
+	}
+
+	cephResourceSpec.Phase = phase
+}
+
+func addStorageRelatedAnnotations(obj client.Object, storageClassClaimName, storageClaim, cephUserType string) {
+	annotations := obj.GetAnnotations()
+	if annotations == nil {
+		annotations = map[string]string{}
+		obj.SetAnnotations(annotations)
+	}
+
+	annotations[StorageClassClaimAnnotation] = storageClassClaimName
+	annotations[controllers.StorageClaimAnnotation] = storageClaim
+	annotations[controllers.StorageCephUserTypeAnnotation] = cephUserType
+}
+
+func (r *StorageClassClaimReconciler) own(resource metav1.Object) error {
+	// Ensure storageClassClaim ownership on a resource
+	return ctrl.SetControllerReference(r.storageClassClaim, resource, r.Scheme)
+}
+
+func (r *StorageClassClaimReconciler) get(obj client.Object) error {
+	key := client.ObjectKeyFromObject(obj)
+	return r.Client.Get(r.ctx, key, obj)
+}
+
+func (r *StorageClassClaimReconciler) update(obj client.Object) error {
+	if err := r.Client.Update(r.ctx, obj); err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	return nil
```

Comment:
This is wrong, If the object is not found we should return an error (this is an update operation, not a delete operation)
```suggestion
	return r.Client.Update(r.ctx, obj);
```

Reply:
ack